sdk/core/System.ServiceModel.Rest/src/Pipeline/HttpPipelineMessageTransport.cs

```diff
@@ -147,23 +145,27 @@ private async ValueTask ProcessSyncOrAsync(PipelineMessage message, bool async)
             }
         }
         // HttpClient on NET5 throws OperationCanceledException from sync call sites, normalize to TaskCanceledException
-        catch (OperationCanceledException e) when (ClientUtilities.ShouldWrapInOperationCanceledException(e, message.CancellationToken))
+        catch (OperationCanceledException e) when (ClientUtilities.ShouldWrapInOperationCanceledException(e, options.CancellationToken))
         {
-            throw ClientUtilities.CreateOperationCanceledException(e, message.CancellationToken);
+            throw ClientUtilities.CreateOperationCanceledException(e, options.CancellationToken);
         }
         catch (HttpRequestException e)
         {
             throw new MessageFailedException(e.Message, e);
         }
 
         OnReceivedResponse(message, responseMessage, contentStream);
+
+        // Set IsError meta-data on the response.
```

Comment:
Design question here: is the transport the best place to set response meta-data?  I kind of think it is, but I'm interested in alternative viewpoints on this.

There's an argument to be made that buffering the response is "meta-data" (or, value-add over the System.Net.Http response we get from HttpClient, and if it's needed by any policy "to the right" of the transport, why wouldn't it live on the transport as well?  Why should the response buffering policy be separate from the transport?)